### PR TITLE
Store Customization > Schedule an action to update the patterns content whenever the business description is changed

### LINKED ISSUE #10659: Store Customization > Schedule an action to update the patterns content whenever the business description is changed
⚠️ This issue can be worked on as soon as https://github.com/woocommerce/woocommerce-blocks/issues/10623 is complete.

Whenever the user changes their business description on the AI settings page, updating the [`'woo_ai_describe_store_description'`](https://github.com/woocommerce/woocommerce/blob/9418072a472b53bd4519086bc1958145600c42ae/plugins/woo-ai/includes/class-woo-ai-settings.php#L39) option, we should schedule an action (via [Action Scheduler](https://actionscheduler.org/)) for triggering the AI content generation in patterns. 

For users who already have this setting filled out prior to the release of the feature developed as part of this project, we should also schedule an action to run just once to take care of these requests and populate the patterns.

### PR DESCRIPTION
**Pull request by nefeline**
## What

Schedules an action whenever the store owner changes their business description is updated in the AI settings screen (whenever the ['woo_ai_describe_store_description'](https://github.com/woocommerce/woocommerce/blob/9418072a472b53bd4519086bc1958145600c42ae/plugins/woo-ai/includes/class-woo-ai-settings.php#L39) option is updated) to update the patterns images and content.

Fixes #10659

## Why

This is a requirement to ensure the images and content in patterns are properly updated whenever the business description is changed.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Install and activate the `woo-ai` plugin (it is available within the [WooCommerce monorepo](https://github.com/woocommerce/woocommerce), under the plugins directory).
2. From your dashboard, head over to WooCommerce > Settings > Advanced > Features
3. Fill out a description for your business and save

Screenshot 2023-09-16 at 09 22 29

5. Now head over to Tools > Scheduled Actions
6. Search for the woocommerce_update_patterns_content hook
7. Make sure the action was successfully scheduled:

Screenshot 2023-09-16 at 09 27 23

8. If the status is still Pending when you check, click on Run and ensure it is successfully executed. If the status is already Complete, no action is required as the process already finished successfully.
9. Check your error log and make sure you don't see any PHP errors.

* [ ] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

N/A

### CODE REVIEW

**Review by albarin — APPROVED** [review]
Works great! 👏

File: src/BlockPatterns.php [hunk 1]
```diff
@@ -198,4 +203,54 @@ public function register_block_patterns() {
...
 		}
 	}
+
+	/**
+	 * Update the patterns content when the store description is changed.
+	 *
+	 * @param string $option The option name.
+	 * @param string $value The option value.
+	 *
+	 * @return bool|int|string|void|\WP_Error
```
Comment (kmanijak) [inline src/BlockPatterns.php]: Minor: I think this function may only return `null`

**Review by kmanijak — APPROVED** [review]
Works well, I left a minor comment but pre-approving!

File: src/BlockPatterns.php (hunk 1 above)
Comment (albarin) [reply src/BlockPatterns.php]: Thanks @kmanijak, you are right, I've removed the @return from the comment.

**Pull request merged**